code/modules/atmospherics/environmental/LINDA_turf_tile.dm

```diff
@@ -144,16 +147,28 @@
 			.[gastype] = TRUE
 
 /////////////////////////////SIMULATION///////////////////////////////////
-
+#ifdef TRACK_MAX_SHARE
 #define LAST_SHARE_CHECK \
 	var/last_share = our_air.last_share;\
+	max_share = max(last_share, max_share);\
 	if(last_share > MINIMUM_AIR_TO_SUSPEND){\
 		our_excited_group.reset_cooldowns();\
 		cached_atmos_cooldown = 0;\
 	} else if(last_share > MINIMUM_MOLES_DELTA_TO_MOVE) {\
 		our_excited_group.dismantle_cooldown = 0;\
 		cached_atmos_cooldown = 0;\
 	}
+#else
+#define LAST_SHARE_CHECK \
+	var/last_share = our_air.last_share;\
+	if(last_share > MINIMUM_AIR_TO_SUSPEND){\
+		our_excited_group.reset_cooldowns();\
+		cached_atmos_cooldown = 0;\
+	} else if(last_share > MINIMUM_MOLES_DELTA_TO_MOVE) {\
+		our_excited_group.dismantle_cooldown = 0;\
+		cached_atmos_cooldown = 0;\
+	}
```

Comment:
```suggestion
#define LAST_SHARE_CHECK \
do { \
	var/last_share = our_air.last_share;\
	if(last_share > MINIMUM_AIR_TO_SUSPEND){\
		our_excited_group.reset_cooldowns();\
		cached_atmos_cooldown = 0;\
	} else if(last_share > MINIMUM_MOLES_DELTA_TO_MOVE) {\
		our_excited_group.dismantle_cooldown = 0;\
		cached_atmos_cooldown = 0;\
	}; \
} while(FALSE)
```

Reply:
Could ya explain the benefit of this? is it to scope the variable?

Reply:
The benefit is slim, less changes of, when using the macro, to incur in improper indentation issues
Also the variables get defined within the macro's scope, instead of on the rest of the proc's
The compiler already checks there's a `do-while(FALSE)` and doesn't even check the `while` at runtime, so there's no loss in performance or anything.